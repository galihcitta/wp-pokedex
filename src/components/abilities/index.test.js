import React from "react";
import Abilities from "./index";
import AbilitiesStyle from "./index.style";

let element;
beforeEach(() => {
    const abilities = [
        {
            ability: {
                name: "chlorophyll",
                url: "https://pokeapi.co/api/v2/ability/34/",
            },
            is_hidden: true,
            slot: 3,
        },
        {
            ability: {
                name: "overgrow",
                url: "https://pokeapi.co/api/v2/ability/65/",
            },
            is_hidden: false,
            slot: 1,
        },
    ];
    element = <Abilities data={abilities}></Abilities>;
});

describe("Abilities Component", () => {
    it("renders correct Abilities and snapshot element <Abilities/>", () => {
        expect(toJson(mount(element))).toMatchSnapshot();
    });

    it("renders correct Abilities and snapshot loading element <Abilities/>", () => {
        expect(shallow(<Abilities.Loading></Abilities.Loading>).exists()).toBe(
            true,
        );
    });

    it("renders correct Abilities and snapshot loading element <Abilities.Loading/>", () => {
        expect(
            toJson(mount(<Abilities.Loading></Abilities.Loading>)),
        ).toMatchSnapshot();
    });
    it("renders correct Abilities and snapshot style element <AbilitiesStyle/>", () => {
        expect(
            toJson(mount(<AbilitiesStyle></AbilitiesStyle>)),
        ).toMatchSnapshot();
    });

    it("check total length element loop <CardMoves/>", () => {
        expect(
            mount(element).find({ "test-id": "box-attribute" }).length,
        ).toEqual(2);
    });

    it("check first element name <CardMoves/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "box-attribute" })
                .first()
                .find("span")
                .first()
                .text(),
        ).toEqual("chlorophyll");
    });

    it("check element have class hidden <CardMoves/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "box-attribute" })
                .first()
                .hasClass("is_hidden"),
        ).toEqual(true);
    });

    it("check element have slot number <CardMoves/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "box-attribute" })
                .first()
                .find("span")
                .last()
                .text(),
        ).toEqual("3");
    });
});
