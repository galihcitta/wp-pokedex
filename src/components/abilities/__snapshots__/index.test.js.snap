// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abilities Component renders correct Abilities and snapshot element <Abilities/> 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 .label-types {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 10px 0;
  width: 100%;
}

.c0 .label-types--box {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 5px;
  margin-right: 5px;
  padding: 7px 15px;
  padding: 10px;
}

.c0 .label-types--box.is_hidden {
  opacity: 0.5;
}

.c0 .label-types--box > span:last-child {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(214,214,214,0.16);
  border: 1px solid #dcdcdc;
  border-radius: 50%;
  color: #4a4a4a;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 22px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 8px;
  width: 22px;
}

<Abilities
  data={
    Array [
      Object {
        "ability": Object {
          "name": "chlorophyll",
          "url": "https://pokeapi.co/api/v2/ability/34/",
        },
        "is_hidden": true,
        "slot": 3,
      },
      Object {
        "ability": Object {
          "name": "overgrow",
          "url": "https://pokeapi.co/api/v2/ability/65/",
        },
        "is_hidden": false,
        "slot": 1,
      },
    ]
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <div
        className="label-types"
      >
        <div
          className="label-types--box is_hidden"
          key="chlorophyll"
          test-id="box-attribute"
        >
          <span>
            chlorophyll
          </span>
           
          <span>
            3
          </span>
        </div>
        <div
          className="label-types--box "
          key="overgrow"
          test-id="box-attribute"
        >
          <span>
            overgrow
          </span>
           
          <span>
            1
          </span>
        </div>
      </div>
    </div>
  </styled.div>
</Abilities>
`;

exports[`Abilities Component renders correct Abilities and snapshot loading element <Abilities.Loading/> 1`] = `
.c1 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 120px;
}

.c2 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 40;
}

.c3 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 150px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 .label-types {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 10px 0;
  width: 100%;
}

.c0 .label-types--box {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 5px;
  margin-right: 5px;
  padding: 7px 15px;
  padding: 10px;
}

.c0 .label-types--box.is_hidden {
  opacity: 0.5;
}

.c0 .label-types--box > span:last-child {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(214,214,214,0.16);
  border: 1px solid #dcdcdc;
  border-radius: 50%;
  color: #4a4a4a;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 22px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 8px;
  width: 22px;
}

<AbilitiesLoading>
  <styled.div>
    <div
      className="c0"
    >
      <div
        className="label-types"
      >
        <div
          className="label-types--box is_hidden"
          test-id="box-attribute"
        >
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="120px"
            >
              <styled.div
                circle={false}
                height={10}
                style={Object {}}
                width="120px"
              >
                <div
                  className="c1"
                  height={10}
                  style={Object {}}
                  width="120px"
                />
              </styled.div>
            </Skeleton>
          </span>
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="40"
            >
              <styled.div
                circle={false}
                height={10}
                style={Object {}}
                width="40"
              >
                <div
                  className="c2"
                  height={10}
                  style={Object {}}
                  width="40"
                />
              </styled.div>
            </Skeleton>
          </span>
        </div>
        <div
          className="label-types--box"
          test-id="box-attribute"
        >
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="150px"
            >
              <styled.div
                circle={false}
                height={10}
                style={Object {}}
                width="150px"
              >
                <div
                  className="c3"
                  height={10}
                  style={Object {}}
                  width="150px"
                />
              </styled.div>
            </Skeleton>
          </span>
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="40"
            >
              <styled.div
                circle={false}
                height={10}
                style={Object {}}
                width="40"
              >
                <div
                  className="c2"
                  height={10}
                  style={Object {}}
                  width="40"
                />
              </styled.div>
            </Skeleton>
          </span>
        </div>
      </div>
    </div>
  </styled.div>
</AbilitiesLoading>
`;

exports[`Abilities Component renders correct Abilities and snapshot style element <AbilitiesStyle/> 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 .label-types {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 10px 0;
  width: 100%;
}

.c0 .label-types--box {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 5px;
  margin-right: 5px;
  padding: 7px 15px;
  padding: 10px;
}

.c0 .label-types--box.is_hidden {
  opacity: 0.5;
}

.c0 .label-types--box > span:last-child {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(214,214,214,0.16);
  border: 1px solid #dcdcdc;
  border-radius: 50%;
  color: #4a4a4a;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 22px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 8px;
  width: 22px;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;
